mysql/mysql-lib.pl

```diff
@@ -1034,7 +1034,7 @@ sub execute_sql_file
 			 : "";
 local $temp = &transname();
 &open_tempfile(TEMP, ">$temp");
-&print_tempfile(TEMP, "source ".$file.";\n");
+&print_tempfile(TEMP, "source ".&fix_collation($file).";\n");
```

Comment:
You will need to make sure that this temp file returned by fix_collation has the same owner as the original file, because sometimes the `mysql` command gets run as the domain owner.

Reply:
Alright, how about that https://github.com/webmin/webmin/pull/1939/commits/b49d4f8147a5427d5ad1f3e9c98a39a75613d58e.